### ISSUE TITLE
Cannot update from 3.3.1 to 3.3.2 within the CP

### ISSUE THREAD

**Issue by ebeauchamps** [statamic/cms#5572]
### Bug description

Composer related error when trying to update from 3.3.1 within my local install (Statamic.test, served through Valet, php 8.1.4)

Prior to this update I ran ```composer global update```in my Terminal

### How to reproduce

Click on Update to the latest version in the CP

### Logs

```shell
./composer.json has been updated
Running composer update statamic/cms --with-dependencies

Fatal error: Uncaught TypeError: Composer\Pcre\Preg::isMatch(): Argument #2 ($subject) must be of type string, bool given, called in /Users/efb/Sites/statamic/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php on line 580 and defined in /Users/efb/Sites/statamic/vendor/composer/pcre/src/Preg.php:233
Stack trace:
#0 /Users/efb/Sites/statamic/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php(580): Composer\Pcre\Preg::isMatch('{(^|:)/Users/ef...', false)
#1 /Users/efb/Sites/statamic/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php(188): Composer\EventDispatcher\EventDispatcher->ensureBinDirIsInPath()
#2 /Users/efb/Sites/statamic/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php(122): Composer\EventDispatcher\EventDispatcher->doDispatch(Object(Composer\Script\Event))
#3 /Users/efb/Sites/statamic/vendor/composer/composer/src/Composer/Installer.php(269): Composer\EventDispatcher\EventDispatcher->dispatchScript('pre-update-cmd', true)
#4 /Users/efb/Sites/statamic/vendor/composer/composer/src/Composer/Command/RequireCommand.php(425): Composer\Installer->run()
#5 /Users/efb/Sites/statamic/vendor/composer/composer/src/Composer/Command/RequireCommand.php(288): Composer\Command\RequireCommand->doUpdate(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput), Object(Composer\IO\ConsoleIO), Array, 'require', 'require-dev')
#6 /Users/efb/Sites/statamic/vendor/symfony/console/Command/Command.php(291): Composer\Command\RequireCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 /Users/efb/Sites/statamic/vendor/symfony/console/Application.php(989): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 /Users/efb/Sites/statamic/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand(Object(Composer\Command\RequireCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /Users/efb/Sites/statamic/vendor/composer/composer/src/Composer/Console/Application.php(332): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /Users/efb/Sites/statamic/vendor/symfony/console/Application.php(171): Composer\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /Users/efb/Sites/statamic/vendor/composer/composer/src/Composer/Console/Application.php(128): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /Users/efb/Sites/statamic/vendor/composer/composer/bin/composer(84): Composer\Console\Application->run()
#13 {main}
  thrown in /Users/efb/Sites/statamic/vendor/composer/pcre/src/Preg.php on line 233
```
```


### Versions

Statamic 3.3.1 Pro
Laravel 9.5.1
PHP 8.1.4
No addons installed

### Installation

Fresh statamic/statamic site via CLI

### Additional details

_No response_

**Comment by dennismenken** [comment]
Same problem on my side/site. I've updated statamic via composer.

**Comment by stephenvoisey** [comment]
Ran into the same issue with 3.3.2 > 3.3.3 update within the CP.

```
./composer.json has been updated
Running composer update statamic/cms --with-dependencies
PHP Fatal error:  Uncaught TypeError: Composer\Pcre\Preg::isMatch(): Argument #2 ($subject) must be of type string, bool given, called in /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php on line 580 and defined in /var/www/shipmate-documentation/vendor/composer/pcre/src/Preg.php:233
Stack trace:
#0 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php(580): Composer\Pcre\Preg::isMatch()
#1 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php(188): Composer\EventDispatcher\EventDispatcher->ensureBinDirIsInPath()
#2 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php(122): Composer\EventDispatcher\EventDispatcher->doDispatch()
#3 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Installer.php(269): Composer\EventDispatcher\EventDispatcher->dispatchScript()
#4 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Command/RequireCommand.php(425): Composer\Installer->run()
#5 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Command/RequireCommand.php(288): Composer\Command\RequireCommand->doUpdate()
#6 /var/www/shipmate-documentation/vendor/symfony/console/Command/Command.php(291): Composer\Command\RequireCommand->execute()
#7 /var/www/shipmate-documentation/vendor/symfony/console/Application.php(989): Symfony\Component\Console\Command\Command->run()
#8 /var/www/shipmate-documentation/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand()
#9 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Console/Application.php(332): Symfony\Component\Console\Application->doRun()
#10 /var/www/shipmate-documentation/vendor/symfony/console/Application.php(171): Composer\Console\Application->doRun()
#11 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Console/Application.php(128): Symfony\Component\Console\Application->run()
#12 /var/www/shipmate-documentation/vendor/composer/composer/bin/composer(84): Composer\Console\Application->run()
#13 /var/www/shipmate-documentation/vendor/bin/composer(117): include('...')
#14 {main}
	thrown in /var/www/shipmate-documentation/vendor/composer/pcre/src/Preg.php on line 233

Fatal error: Uncaught TypeError: Composer\Pcre\Preg::isMatch(): Argument #2 ($subject) must be of type string, bool given, called in /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php on line 580 and defined in /var/www/shipmate-documentation/vendor/composer/pcre/src/Preg.php:233
Stack trace:
#0 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php(580): Composer\Pcre\Preg::isMatch()
#1 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php(188): Composer\EventDispatcher\EventDispatcher->ensureBinDirIsInPath()
#2 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/EventDispatcher/EventDispatcher.php(122): Composer\EventDispatcher\EventDispatcher->doDispatch()
#3 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Installer.php(269): Composer\EventDispatcher\EventDispatcher->dispatchScript()
#4 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Command/RequireCommand.php(425): Composer\Installer->run()
#5 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Command/RequireCommand.php(288): Composer\Command\RequireCommand->doUpdate()
#6 /var/www/shipmate-documentation/vendor/symfony/console/Command/Command.php(291): Composer\Command\RequireCommand->execute()
#7 /var/www/shipmate-documentation/vendor/symfony/console/Application.php(989): Symfony\Component\Console\Command\Command->run()
#8 /var/www/shipmate-documentation/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand()
#9 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Console/Application.php(332): Symfony\Component\Console\Application->doRun()
#10 /var/www/shipmate-documentation/vendor/symfony/console/Application.php(171): Composer\Console\Application->doRun()
#11 /var/www/shipmate-documentation/vendor/composer/composer/src/Composer/Console/Application.php(128): Symfony\Component\Console\Application->run()
#12 /var/www/shipmate-documentation/vendor/composer/composer/bin/composer(84): Composer\Console\Application->run()
#13 /var/www/shipmate-documentation/vendor/bin/composer(117): include('...')
#14 {main}
	thrown in /var/www/shipmate-documentation/vendor/composer/pcre/src/Preg.php on line 233
```

**Comment by jasonvarga** [comment]
You'll have to use the command line to update at the moment.
I've opened an issue on the Composer repo.

https://github.com/composer/composer/issues/10662

**Comment by jasonvarga** [comment]
This has been fixed in Composer and will be available in their next 2.3.x release.
